src/components/map/draw-tools/DrawingToolbox.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react';
+
+import { Feature, Map } from 'ol';
+import { GeoJSON } from 'ol/format';
+import { Circle, Geometry } from 'ol/geom';
+import { Type } from 'ol/geom/Geometry';
+import { fromCircle } from 'ol/geom/Polygon';
+import { Draw, Interaction, Modify, Snap } from 'ol/interaction';
+import { defaults } from 'ol/interaction/defaults';
+import { Options, createBox } from 'ol/interaction/Draw';
+import VectorSource from 'ol/source/Vector';
+import { PiCircle, PiPolygonFill, PiRectangle } from 'react-icons/pi';
+
+import { DATA_PROJECTION, MAP_PROJECTION } from '@/components/Map';
+import { useFilters } from '@/hooks/useFilters';
+
+import { DrawingTool } from './DrawingTool';
+
+import './DrawingToolbox.scss';
+
+type DrawingToolboxProps = {
+  isToolboxVisible: boolean;
+  map: Map;
+  drawingSource: VectorSource<Feature<Geometry>>;
+};
+
+enum Shapes {
+  CIRCLE = 'Circle',
+  BOX = 'box',
+  POLYGON = 'Polygon',
+}
+
+export const DrawingToolbox = ({ isToolboxVisible, map, drawingSource }: DrawingToolboxProps) => {
+  const {
+    actions: { setAoiFilter },
+  } = useFilters();
+
+  const [isActive, setIsActive] = useState('');
+
+  const drawShape = (type: Type | string) => {
+    setIsActive(type);
+
+    // I don't know if this is a problem with how I've coded this but if you add an interaction
```

Comment:
Probably a render issue. There's always going to be issues when handling the OL and React state together.